lisa/trace.py

```diff
@@ -91,10 +91,8 @@ def add_events_deltas(self, df, col_name='delta', inplace=True):
         if not inplace:
             df = df.copy()
 
-        df.loc[df.index[:-1], col_name] = df.index.values[1:] - df.index.values[:-1]
-        # Fix the last event, which will have a NaN duration
-        # Set duration to trace_end - last_event
-        df.loc[df.index[-1], col_name] = self.end - df.index[-1]
+        df[col_name] = df.index
+        df[col_name] = df[col_name].diff()
```

Comment:
AFAICT to maintain current behaviour this should be 
```suggestion
        df[col_name] = df[col_name].diff().shift(-1)
```

And you need to keep the `df.loc[df.index[-1], col_name]` fixup